### PR TITLE
coq_8_12: init at 8.12+β1

### PR DESCRIPTION
**Pull request by vbgl**
###### Motivation for this change

https://coq.github.io/doc/V8.12+beta1/refman/changes.html#version-8-12

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/coq-modules/mathcomp/default.nix [hunk 1]
```diff
@@ -76,7 +76,7 @@ let
     };
     # versions of coq compatible with released mathcomp versions
     coq-versions     = {
-      "1.11.0"       = flip elem [ "8.7" "8.8" "8.9" "8.10" "8.11" ];
+      "1.11.0"       = flip elem [ "8.7" "8.8" "8.9" "8.10" "8.11" "8.12" ];
       "1.11+beta1"   = flip elem [ "8.7" "8.8" "8.9" "8.10" "8.11" ];
```
Comment (Zimmi48) [inline pkgs/development/coq-modules/mathcomp/default.nix:80]: I believe that 1.11+beta1 is compatible with Coq 8.12 as well. But do you really want to keep this beta version in the list of available versions? We usually remove the beta versions of Coq for instance...

Comment (vbgl) [reply pkgs/development/coq-modules/mathcomp/default.nix:80]: I don’t know and I didn’t check whether the β versions are compatible.

Maybe Cyril has an opinion about this. (ping @CohenCyril)

Comment (Zimmi48) [reply pkgs/development/coq-modules/mathcomp/default.nix:80]: FWIW, I didn't check in Nix, but I did check in Coq's v8.12 CI before 1.11.0 was released.

Comment (vbgl) [reply pkgs/development/coq-modules/mathcomp/default.nix]: I’ve just added “8.12” to the “1.11+beta1” list.

Whether said list should be kept or not is a different topic, slightly independent from this PR.

Comment (CohenCyril) [reply pkgs/development/coq-modules/mathcomp/default.nix]: I approve of this change.

**Review by CohenCyril — APPROVED** [review]

**Pull request merged**